src/common/storage/src/config.rs

```diff
@@ -474,3 +478,60 @@ impl ShareTableConfig {
         }
     }
 }
+
+#[derive(Clone, Default, PartialEq, Eq, Debug, Serialize, Deserialize)]
+#[serde(default)]
+pub struct CatalogConfig {
+    #[serde(flatten)]
+    pub catalogs: HashMap<String, CatalogDescription>,
```

Comment:
Catalogs should be placed outside. `common-storage` only need to hold something like `StorageParams`.

Reply:
Then a new `common-extern-catalog` module in `src/common` may be needed.🤔

Cannot use `common-catalog` because it conflicts with `src/query/catalog`.

Reply:
Maybe we can place it in `src/query/catalog` directly.

Reply:
> Catalogs should be placed outside. `common-storage` only need to hold something like `StorageParams`.

Oh, here is some misunderstanding.

I mean, we should not provide:

```rust
pub struct CatalogConfig {
    #[serde(flatten)]
    pub catalogs: HashMap<String, CatalogDescription>,
}
```

Instead, we need to provide `CatalogDescription` directly without a `hashmap`. Users need to build their own `HashMap` while needed.

After this change, maybe we can rename `CatalogDescription` to `CatalogConfig`?

Reply:
> Maybe we can place it in `src/query/catalog` directly.

`common-config` cannot depend on `common-config` (`src/query/catalog`), or there will be a circled dependency.

Reply:
> `common-config` cannot depend on `common-catalog` (`src/query/catalog`), or there will be a circled dependency.

We can place it at `query/config/src/inner`. `common-storage` is not a good place.